astropy/nddata/nduncertainty.py

```diff
@@ -748,6 +782,13 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
     def _data_unit_to_uncertainty_unit(self, value):
         return value
 
+    def _convert_to_variance(self):
+        return VarianceUncertainty(self.array ** 2, unit=self.unit ** 2)
```

Comment:
This fails because it is possible for `self.unit` to be `None`: https://github.com/astropy/astropy/runs/6088416154?check_suite_focus=true#step:6:1500

Reply:
`data` can also be None when instantiating an uncertainty object, so I guess this should also be handled here.